homeassistant/components/rointe/device_manager.py

```diff
@@ -0,0 +1,384 @@
+        return None
+
+    product_type = device_data["data"].get("type", None)
+    version = device_data["data"].get("product_version", None)
+    current_firmware = device_data["firmware"].get("firmware_version_device", None)
+
+    if not product_type or not version or not current_firmware:
+        LOGGER.warning(
+            "Unable to determine latest FW for [%s][%s] at v[%s]",
+            product_type,
+            version,
+            current_firmware,
+        )
+        return None
+
+    product = get_product_by_type_version(product_type, version)
+
+    if not product:
+        LOGGER.warning(
+            "Product not found: [%s][%s] at v[%s]",
+            product_type,
+            version,
+            current_firmware,
+        )
+        return None
+
+    if product in fw_map and version in fw_map[product]:
+        return fw_map[product][version]
+
+    # If no update path available return the current firmware string.
+    return current_firmware
+
+
+class RointeDeviceManager:
+    """Device Manager."""
+
+    def __init__(
+        self,
+        username: str,
+        password: str,
+        installation_id: str,
+        hass: HomeAssistant,
+        rointe_api: RointeAPI,
+    ) -> None:
+        """Initialize the device manager."""
+        self.username = username
+        self.password = password
+        self.installation_id = installation_id
+        self.rointe_api = rointe_api
+
+        self.hass = hass
+        self.auth_token = None
+        self.auth_token_expire_date: datetime | None = None
+
+        self.rointe_devices: dict[str, RointeDevice] = {}
+
+    def _fail_all_devices(self):
+        """Set all devices as unavailable."""
+
+        if self.rointe_devices:
+            for device in self.rointe_devices.values():
+                device.hass_available = False
+
+    async def update(self) -> dict[str, list[RointeDevice]]:
+        """Retrieve the devices from the user's installation."""
+
+        LOGGER.debug("Device manager updating")
+
+        installation_response: ApiResponse = await self.hass.async_add_executor_job(
+            self.rointe_api.get_installation_by_id, self.installation_id
+        )
+
+        if not installation_response.success:
+            LOGGER.error(
+                "Unable to get Rointe Installations. Error: %s",
+                installation_response.error_message,
+            )
+            self._fail_all_devices()
+            return {}
+
+        installation = installation_response.data
+        discovered_devices: dict[str, list[RointeDevice]] = {}
+        firmware_map = await self._get_firmware_map()
+
+        for zone_key in installation["zones"]:
+            LOGGER.debug("Processing zone: %s", zone_key)
+
+            zone = installation["zones"][zone_key]
+
+            if "devices" not in zone:
+                LOGGER.debug("No devices info found for zone")
+                continue
+
+            devices = zone["devices"]
```

Comment:
This check can be simplified:

```python
if (devices := zone.get("devices")) is None:
    LOGGER.debug("No devices info found for zone")
    continue
```

After this point, `devices` will contain what you need?